MOV-AI/containers-ign-simulator
# Feature/dp 1320 add smoke tests to simulator

## Content
- new feature branch to add a Web API to our IGN Gazebo Simulation container

## Steps to test functionality:
- build container: `docker build -t tugbot-ign-sim:local -f gazebo/Dockerfile .`
- run container: 
```
docker run --rm  -td \
--env="DISPLAY=:1" \
--env="QT_X11_NO_MITSHM=1" \
--env="NVIDIA_VISIBLE_DEVICES=all" \
--env="NVIDIA_DRIVER_CAPABILITIES=all" \
--volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" \
--volume="$XAUTH:$XAUTH" \
--runtime=nvidia \
--network=host \
--privileged \
--env="IGN_IP=$(hostname -I | cut -f 1 -d ' ')" \
--env="IGN_GAZEBO_SYSTEM_PLUGIN_PATH=/movai_ign_plugins/system" \
--env="IGN_GUI_PLUGIN_PATH=/movai_ign_plugins/gui" \
--env="IGN_GAZEBO_RESOURCE_PATH=/opt/mov.ai/app/models/movai_sim_tugbot/worlds:/opt/mov.ai/app/models/movai_sim_tugbot/models:/opt/mov.ai/app/models/movai_sim_models/models:/opt/mov.ai/app/models/movai_sim_models/worlds:/opt/mov.ai/app/models/movai_sim_models:/opt/mov.ai/app/models/movai_sim_tugbot:/models_shared"  \
tugbot-ign-sim:local \
ign gazebo empty.sdf -r
```

- start publishing topic /test_from_spawner in host:
```
duration=60; end=$((SECONDS+duration)); while [ $SECONDS -lt $end ]; do ign topic -p "data:\"test\"" -t /test_from_spawner --msgtype ignition.msgs.StringMsg; done &
```

- start echoing topic /test_from_sim in host

- do a post request: `POST http://127.0.0.1:8081/communication-test`
- check `docker logs pokemongym-simulator -f`

---------------

Still missing:
- html page to document endpoints (swagger)
- configuration endpoint
- simulation-instance endpoint

What will be necessary on movai-service:
- healthcheck for celery and rabbitmq and the api


---------------

- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes


[^note]:
    Put an `x` into the [ ] to show you have filled the information.
    The template comes from https://github.com/MOV-AI/.github/blob/master/.github/pull_request_template.md
    You can override it by creating .github/pull_request_template.md  in your own repository
